### PR TITLE
refactor!: remove not core modules

### PR DESCRIPTION
**Pull request by dongsam**
### Remove Canto Additional Ethermint Dependent Modules

- [x]  `CSR`

### Remove IBC Modules

- [x]  `IBC`
- [x]  `Transfer`

### Remove Cosmos SDK Optional Modules

- [x]  `Authz`

### Remove Canto Additional Ethermint/IBC Dependent Modules

- [x]  `Onboarding`
- [x]  `Coinswap`

## TODO

- [x] re-generation proto, *.pb.go, Swagger 
- [x] check simulations

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] targeted the correct branch (see [PR Targeting](https://github.com/canto-network/canto/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] included the necessary unit and integration tests
- [ ] reviewed "Files changed" and left comments if necessary
 
### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/erc20/types/msg.go
Comment (jasonsong0) [inline x/erc20/types/msg.go:1]: it seems that ```ibctransfertypes.ValidateIBCDenom``` is not unnecessary.

**Review by jasonsong0 — COMMENTED** [review]
LGTM apart from a comment above.
Is client doc automatically  generated? If not, it seems better to delete the docs for deleted modules

File: x/erc20/types/msg.go
Comment (dongsam) [reply x/erc20/types/msg.go:1]: Yes, this was still a PR under WIP, thank you for the interim review and I will ask you for a review request again when it is R4R

Comment (dongsam) [reply x/erc20/types/msg.go:1]: I just fixed it on the latest commits. Could you interim review again?

**Review by jasonsong0 — APPROVED** [review]
LGTM

**Pull request merged**